src/main/kotlin/com/github/treepat/match_manager/MatchManager.kt

```diff
@@ -0,0 +1,64 @@
+package com.github.treepat.match_manager
+
+import com.github.treepat.expression.TreepatExpression
+import com.github.treepat.target_tree.TargetTree
+import com.github.treepat.target_tree.TargetTreeNode
+
+class MatchManager(private var expression: TreepatExpression, private var tree: TargetTree) {
```

Comment:
The previous solution (run any `TreepatExpression` over any `TargetTree` ) was great because it was flexible to the user. However, if we wanted to get the matches one by one it was necessary to create a class that used both of them (As it needed to remember exactly which node of a specific tree had to be evaluated).

This is how MatchManager was born (name may be subject to changes), which combines a `TargetTree` and a `TreepatExpression` and has the next important methods:

Reply:
Great idea! I have a question, in the constructor we have `var`. Can. we change `var` to `val` as these properties don't seem to be changing after the object is instantiated.

Reply:
@johan-smc Even though i forgot to state it, the idea of using `MatchManager` was deprecating all `TargetTree` functions. However we could try to put a `HashMap` inside `TreeTarget` that has a `MatchManager` for each `TreepatExpression`, what do you think?

Reply:
Mmm No, a `HashMap` is not necessary in this case,  maybe only have one  `MatchManage` is needed when you want to get a single response in parts, and create other function that inits this variable sending the `TreepatExpression`in params. The other functions can accept the `TreepatExpression` in param and create a `MatchManage` in the scope of the functions. But I'm not sure what it is better, we need to have a discussion about that. 👍